lock/src/main/java/com/auth0/android/lock/views/DomainFormView.java

```diff
@@ -0,0 +1,243 @@
+        usernameInput.setVisibility(View.GONE);
+        usernameInput.clearInput();
+        actionButton.setEnabled(false);
+        actionButton.setText(R.string.com_auth0_lock_action_login);
+    }
+
+    private String extractDomain(String email) {
+        int indexAt = email.indexOf("@") + 1;
+        if (indexAt == 0) {
+            return "";
+        }
+        int indexDot = email.indexOf(".", indexAt);
+        String domain;
+        if (indexDot == -1) {
+            domain = email.substring(indexAt);
+        } else {
+            domain = email.substring(indexAt, indexDot);
+        }
+        if (domain.isEmpty()) {
+            return "";
+        }
+        return domain;
+    }
+
+    private String extractUsername(String email) {
+        int indexAt = email.indexOf("@");
+        if (indexAt == -1) {
+            return "";
+        }
+        return email.substring(0, indexAt);
+    }
+
+    public String getUsernameOrEmail() {
+        return usernameInput.getText();
+    }
+
+    public String getPassword() {
+        return passwordInput.getText();
+    }
+
+
+    @Override
+    public void onClick(View v) {
+        if (!hasValidData()) {
+            return;
+        }
+
+        if (passwordInput.getVisibility() == VISIBLE || !currentStrategy.isResourceOwnerEnabled()) {
+            super.onClick(v);
+        } else {
+            goBackBtn.setVisibility(VISIBLE);
+            passwordInput.setVisibility(View.VISIBLE);
+            usernameInput.setVisibility(VISIBLE);
+            if (usernameEmailValidation == ValidatedInputView.DataType.USERNAME) {
+                usernameInput.setText(currentUsername);
+            } else {
+                usernameInput.setText(emailInput.getText());
+            }
+            emailInput.setVisibility(GONE);
+        }
+    }
+
+    @Override
+    protected Object getActionEvent() {
+        if (currentStrategy.isResourceOwnerEnabled()) {
+            return new EnterpriseROLoginEvent(currentStrategy.getConnections().get(0).getName(), getUsernameOrEmail(), getPassword());
+        } else {
+            return new EnterpriseWebLoginEvent(currentStrategy.getConnections().get(0).getName());
+        }
+    }
+
+    @Override
+    protected boolean hasValidData() {
+        boolean valid = true;
+        if (emailInput.getVisibility() == VISIBLE) {
+            valid = emailInput.validate();
+        }
+        if (usernameInput.getVisibility() == VISIBLE) {
+            valid = valid && usernameInput.validate();
+        }
+        if (passwordInput.getVisibility() == VISIBLE) {
+            valid = valid && passwordInput.validate();
+        }
+        return valid;
+    }
+
+    @Nullable
+    private Strategy searchDomain(String domain) {
+        if (domain.isEmpty()) {
+            return null;
+        }
+        for (Strategy s : filteredEnterpriseStrategies) {
+            if (s.getType() == Strategies.Type.ENTERPRISE && domain.toLowerCase().contains(s.getName())) {
```

Comment:
This is not right, the domains are contained in the connection values, the name of the strategy is something different